webpack/assets/javascripts/jquery.ui.custom_spinners.js

```diff
@@ -0,0 +1,115 @@
+const megabyte = 1024 * 1024;
+const gigabyte = 1024 * megabyte;
+
+$(function () {
+  $.widget('ui.limited_spinner', $.ui.spinner, {
+    options: {
+      softMaximum: 0,
+      errorTarget: null
+    },
+    _validate: function () {
+      if (this.options.softMaximum !== 0) {
+        if (this.value() > this.options.softMaximum) {
+          this.options.errorTarget.show();
+        } else {
+          this.options.errorTarget.hide();
+        }
+      }
+    },
+    _spin: function (step, event) {
+      let result = this._super(step, event);
+
+      this._validate();
+      return result;
+    }
+  });
+
+  $.widget('ui.byte_spinner', $.ui.limited_spinner, {
```

Comment:
Please use camel case for functions even if they are defined using a factory which makes linting miss that